### DISCUSSION TITLE
[v3] Injecting nprogress css violates nonce-based CSP-policies

### DISCUSSION THREAD

**Discussion by joostdebruijn** [livewire/livewire#6588] [7. Bugs (archived)] [answered]
### Livewire version

3.0.1

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

When using Livewire v3 in an application with a nonce-based CSP policy, a CSP-violation is triggered. The Bar-plugin injects the nprogress css stylesheet without a nonce: https://github.com/livewire/livewire/blob/ad2caa6da4249fcd895ce6378fca19fad1d8843f/js/plugins/navigate/bar.js#L78

For now I solved it by adding the sha256 hash to the CSP, but that can cause problems when the nprogress stylesheet will be updated.

I'm willing to prepare a pull request to solve this and I see two possible solutions:

- A solution as described in this Intertia-issue: https://github.com/inertiajs/progress/issues/21#issuecomment-1176231789
- A solution as used by FullCalendar: https://fullcalendar.io/docs/content-security-policy#dynamically-generated-styles

The solution as provided by FullCalendar is the one I liked the most as it "just works", without any additional actions needed by the developer. Is there any preference for a solution by the maintainers and/or community members? I would like to know that before I start working on a PR.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by HDVinnie** [comment]
Also facing this issue

**Comment by joostdebruijn** [comment]
I've created a PR for this some time ago: https://github.com/livewire/livewire/pull/6625. It's awaiting merge.

**Comment by piotrku** [comment]
The same here, fingers crossed for the PR to be merged soon.
 Thanks @joostdebruijn for your engagement!

**Comment by joostdebruijn** [comment, marked as answer]
It has been merged and has been released in v3.4.0!
